src/Microsoft.OData.Client/ALinq/ResourceBinder.cs

```diff
@@ -1061,7 +1061,16 @@ private static Expression AnalyzeAddCountOption(MethodCallExpression mce, CountO
 
             ValidationRules.RequireCanAddCount(rse);
             rse.ConvertKeyToFilterExpression();
-            rse.CountOption = countOption;
+
+            ConstantExpression countQueryOption = StripTo<ConstantExpression>(mce.Arguments[0]);
+            if ((bool)countQueryOption.Value == true)
+            {
+                rse.CountOption = CountOption.CountQueryTrue;
+            }
+            else
+            {
+                rse.CountOption = CountOption.CountQueryFalse;
+            }

```

Comment:
What about when count option is neither true or false? i.e. when it should not be explicitly set in the query?

Reply:
If it's not set in the query, the call doesn't reach here

Reply:
`IncludeCount(false)` test case is included. #Resolved